TrackingTools/TrackFitters/plugins/KFFittingSmoother.h

```diff
@@ -126,13 +126,13 @@ namespace {
 #define PRINT LogTrace("")
 #endif
 
-  Trajectory KFFittingSmoother::fitOne(const Trajectory& t, fitType type) const {
```

Comment:
it looks like everything in this .h file is in `namespace {`
this is from https://github.com/cms-sw/cmssw/pull/11931/files#diff-c4c98c70cb79975a056732116a2e860ace3f05d281f555f314f1faa810aaaca5R22

https://github.com/cms-sw/cmssw/commit/1970060c4ef0d01e23074755487272916a79c4aa

@VinInn was it intentional?

Reply:
absolutely.
in plugins everything should be either inlined or in anonymous namespace
1) not to pollute the global namespace
2) to make the symbols "hidden" (no trampolines or other overhead of shared-libs)

Reply:
OK, somehow when I noticed two cases of include in `git grep KFFittingSmoother.h`, I did not notice that the other one was in a non-existent file (now dropped via #34179).

I see that we have  in TrackingTools/TrackFitters/plugins/KFFittingSmootherESProducer.cc
```C++
// to be included only here...
#include "KFFittingSmoother.h"
```

is there a way to actually enforce this with perhaps an ifdef of sorts?
Otherwise, nothing seems to prevent for this file to be included somewhere else.

Reply:
cut&paste?

Reply:
> cut&paste?

I'm not sure what you meant.
Are you suggesting to merge the content of KFFittingSmoother.h into KFFittingSmootherESProducer.cc ?

Reply:
if not used anywhere else

Reply:
it looks like this file was not updated. From the way it is written, we either need a single include protection with e.g. a `#define KFFittingSmootherESProducer_cc` in KFFittingSmootherESProducer.cc and then an ifdef in this file; or move the contents to KFFittingSmootherESProducer.cc

Reply:
ok, I will move its contects in to `KFFittingSmootherESProducer.cc`